hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx

```diff
@@ -1082,76 +1081,98 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
-                        f"(Rate adjusted: {effective_hedging_price_adjusted:.2f} {market_pair.taker.quote_asset})."
+            # Check if bid size is greater than zero
+            if bid_size > s_decimal_zero:
+
+                bid_price = self.c_get_market_making_price(market_pair, True, bid_size)
+
+                if bid_price is not None:
+                    effective_hedging_price = self.c_calculate_effective_hedging_price(
+                        market_pair,
+                        True,
+                        float(bid_size)
                     )
-                self.c_buy_with_specific_market(
-                    market_pair,
-                    bid_size,
-                    order_type=OrderType.LIMIT,
-                    price=bid_price
-                )
+                    effective_hedging_price_adjusted = self._exchange_rate_conversion.adjust_token_rate(
+                        market_pair.taker.quote_asset, effective_hedging_price
+                    )
+                    if self._logging_options & self.OPTION_LOG_CREATE_ORDER:
+                        self.log_with_clock(
+                            logging.INFO,
+                            f"({market_pair.maker.trading_pair}) Creating limit bid order for "
+                            f"{bid_size} {market_pair.maker.base_asset} at "
+                            f"{bid_price} {market_pair.maker.quote_asset}. "
+                            f"Current hedging price: {effective_hedging_price} {market_pair.taker.quote_asset} "
+                            f"(Rate adjusted: {effective_hedging_price_adjusted:.2f} {market_pair.taker.quote_asset})."
+                        )
+                    self.c_buy_with_specific_market(
+                        market_pair,
+                        bid_size,
+                        order_type=OrderType.LIMIT,
+                        price=bid_price
+                    )
+                else:
+                    if self._logging_options & self.OPTION_LOG_NULL_ORDER_SIZE:
+                        self.log_with_clock(
+                            logging.WARNING,
+                            f"({market_pair.maker.trading_pair})"
+                            f"Order book on taker is too thin to place order for size: {bid_size}"
+                            f"Reduce order_size_portfolio_ratio_limit"
+                        )
             else:
                 if self._logging_options & self.OPTION_LOG_NULL_ORDER_SIZE:
                     self.log_with_clock(
                         logging.WARNING,
-                        f"({market_pair.maker.trading_pair})"
-                        f"order book on taker is too thin to place order for size: {bid_size}"
-                        f"Reduce order_size_portfolio_ratio_limit"
+                        f"({market_pair.maker.trading_pair}) Attempting to place a limit bid but the "
+                        f"bid size is 0. Skipping."
```

Comment:
It would be better to show the reason for zero size bid if possible. Users wouldn't know why the bid size is zero.